src/EditorFeatures/CSharp/StringCopyPaste/StringInfo.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Collections;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.StringCopyPaste;
+
+using static StringCopyPasteHelpers;
+
+internal readonly struct StringInfo
```

Comment:
this is information we were computing before, but storing as separate fields.  i aggregated into into a single type to keep things cleaner.